src/Bicep.LangServer/Telemetry/TelemetryConstants.cs

```diff
@@ -16,6 +16,8 @@ public static class EventNames
             public const string ObjectBodySnippetInsertion = nameof(ObjectBodySnippetInsertion);
 
             public const string DisableNextLineDiagnostics = nameof(DisableNextLineDiagnostics);
+            public const string DisableRuleInBicepConfig = nameof(DisableRuleInBicepConfig);
+            public const string OverallLinterStateChangeInBicepConfig = nameof(OverallLinterStateChangeInBicepConfig);
```

Comment:
>OverallLinterStateChangeInBicepConfig [](http://example.com/codeflow?start=32&length=37)

These two new events will tell us about state changes, but don't we need more data to figure if the state changes are significant? If we find out that there are 332 rule changes in bicep config for a specific rule, the conclusion from that would be different if this is across 332 config files vs 1 million.

Reply:
@majastrz , as discussed offline, this only covers text change scenario in bicepconfig.json i.e. user updates the state manually.

We have a follow up task to capture the state of bicepconfig.json on file open:
https://github.com/Azure/bicep/issues/5301

Reply:
Sounds good.